src/main.py

```diff
@@ -38,69 +39,147 @@ def get_current_metric_definitions():
+    six_days_before = latest - pd.Timedelta(days=6)
+    seven_days_before = latest - pd.Timedelta(days=7)
+    # As above
+    thirteen_days_before = latest - pd.Timedelta(days=13)
+
+    area_data_last_week = area_data[area_data.date.ge(six_days_before)]
+    area_data_week_before_last = area_data[area_data.date.ge(thirteen_days_before) & area_data.date.le(seven_days_before)]
+
+    aggregation_output_last_week = getattr(area_data_last_week[metric_name], aggregation_function)()
+    aggregation_output_week_before = getattr(area_data_week_before_last[metric_name], aggregation_function)()
+
+    percentage_change = ((aggregation_output_last_week - aggregation_output_week_before) / aggregation_output_week_before) * 100.0
+    return {
+        "aggregation_output_week_before": {
+            "column_name": f'{metric_name}-{thirteen_days_before.strftime("%m-%d-%Y")}-to-{seven_days_before.strftime("%m-%d-%Y")}',
+            "value": aggregation_output_week_before
+        },
+        "aggregation_output_last_week": {
+            "column_name": f'{metric_name}-{six_days_before.strftime("%m-%d-%Y")}-to-{latest.strftime("%m-%d-%Y")}',
+            "value": aggregation_output_last_week
+        },
+        "percentage_change": {
+            "column_name": 'percentageChange',
+            "value": percentage_change
+        }
+    }
+
+def get_combined_stats(url, metric_name, aggregation_function):
+    metric_df = pd.read_csv(url)
+    metric_df['date'] = pd.to_datetime(metric_df['date'])
+
+    area_names = metric_df.areaName.unique()
+    ret = []
 
-def get_areas_above_threshold(area_type, metric_name, threshold, aggregationFunction):
+    if metric_name == "newCasesBySpecimenDate":
+        populations_df = get_populations()
+        for area_name in area_names:
+            percentage_change_stats = get_percentage_change(area_name, metric_name, metric_df, aggregation_function)
+            area_code = metric_df[metric_df.areaName.eq(area_name)].areaCode.unique()[0]
+            per_100000_population = metric_value_per_100000(populations_df, percentage_change_stats["aggregation_output_last_week"]["value"], area_name, area_code)
+            ret.append([
+                area_name,
+                percentage_change_stats["aggregation_output_week_before"]["value"],
+                percentage_change_stats["aggregation_output_last_week"]["value"],
+                per_100000_population,
+                percentage_change_stats["percentage_change"]["value"],
+            ])
+            column_names = [
+                'areaName',
+                percentage_change_stats["aggregation_output_week_before"]["column_name"],
+                percentage_change_stats["aggregation_output_last_week"]["column_name"],
+                'lastSevenDaysPer100000',
+                percentage_change_stats["percentage_change"]["column_name"],
+            ]
+
+            ret_df = pd.DataFrame(ret, columns = column_names)
+        return ret_df
+    else:
```

Comment:
Repetition here in constructing a table which is the same as above bar one column, which I think I prefer to repeating the `if metric_name == "newCasesBySpecimenDate": ... else: ...` a few times, but I'm open to other suggestions!

Reply:
what about building up the main columns first, then appending a column for cases/100K if relevant? might be slightly more annoying to do when it's in array form, but once it's a Panda DataFrame I think it's simpler: https://www.geeksforgeeks.org/adding-new-column-to-existing-dataframe-in-pandas/

Reply:
and would be nice to take the "generate cols" and "generate new cases col" into separate functions just to keep this one nice and short

Reply:
Good ideas, I was trying to avoid looping through all the area_names a second time for the cases/100K column, but perhaps it would be better like that if it was more readable. Let me have a go and see how it looks

Reply:
I kept just the one loop through the `area_names`, but changed the table construction to use multiple if statements checking for `metric_name == "newCasesBySpecimenDate"` in the end to try to reduce repetition (e.g. line 138 [here](https://github.com/guardian/uk-coronavirus-data-alerts/pull/4/commits/8b9b7af94e5243d138e35b17898f0025b914239f))